### PR TITLE
Add "tags" field to ConnectionStatus

### PR DESCRIPTION
**Pull request by subpop**

### CODE REVIEW

**Comment by subpop** [comment]
@dehort, here's an initial take on a "tags" field in the ConnectionStatus message.

**Review by dehort — COMMENTED** [review]
@subpop I think inventory requires that the tag's keys and values are strings.  So we should probably require the keys / tags to be strings here as well.

**Comment by subpop** [comment]
@dehort Good point. I made sure to explicitly declare that.

**Review by lphiri — COMMENTED** [review]
@subpop @dehort  what namespace in Inventory would these tags live  ("insights-client" ? or something else.
Also how would the user set these tags on the rhc client side?

**Comment by subpop** [comment]
I don't know what namespace we want to put them, but they should be separate from the "insights-client" namespace. `rhcd` the client program is unaware of the concept of namespaced tags, just like `insights-client`. As far as its concerned, its tags are defined in a file, similarly to `insights-client`. You can see more about `rhcd`'s tagging implementation here. https://github.com/RedHatInsights/yggdrasil#tags and https://github.com/RedHatInsights/yggdrasil/commit/f1b60831020a07052ccba22c2e3f3c884f34ba02

**Comment by dehort** [comment]
@lphiri @subpop The client id is being stored as "rhc_client_id" in inventory, so I'm leaning towards using "rhc_client" as the namespace.  Sound reasonable?

File: README.md [hunk 1]
```diff
@@ -122,7 +127,11 @@ A complete example of a `ConnectionStatus` message:
             },
             "echo": {}
         },
-        "state": "online"
+        "state": "online",
+        "tags": {
+            "region": "us1",
+            "prod": false
```
Comment (dehort) [inline README.md:133]: We need to wrap the false in double quotes, but I can do that after its merged.

**Review by dehort — APPROVED** [review]

**Comment by lphiri** [comment]
> @lphiri @subpop The client id is being stored as "rhc_client_id" in inventory, so I'm leaning towards using "rhc_client" as the namespace. Sound reasonable?

That works for me.

**Comment by lphiri** [comment]
> I don't know what namespace we want to put them, but they should be separate from the "insights-client" namespace. `rhcd` the client program is unaware of the concept of namespaced tags, just like `insights-client`. As far as its concerned, its tags are defined in a file, similarly to `insights-client`. You can see more about `rhcd`'s tagging implementation here. https://github.com/RedHatInsights/yggdrasil#tags and [RedHatInsights/yggdrasil@f1b6083](https://github.com/RedHatInsights/yggdrasil/commit/f1b60831020a07052ccba22c2e3f3c884f34ba02)

@subpop  this is ok for a start.  I do think from a UXD perspective it will be easier for  centralized tag setting on the client side, but let's cross that bridge when we get to it :)

**Comment by subpop** [comment]
> @subpop this is ok for a start. I do think from a UXD perspective it will be easier for centralized tag setting on the client side, but let's cross that bridge when we get to it :)

I agree. The data has to get stored somewhere. If we build up software that can automatically write out a `tags.toml`, that's just building on this foundation.

**Review by lphiri — APPROVED** [review]

**Pull request merged**